### PR TITLE
Add Bitrix24 Webhook notification

### LINKED ISSUE #3610: Bitrix24 Notification
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

New Notification

### 🔖 Feature description

I would like it to be added so that we receive notifications through the Bitrix24 system.

### ✔️ Solution

A system that sends notifications can be added using the Bitrix24 webhook.

### ❓ Alternatives

_No response_

### 📝 Additional Context

_No response_

### PR DESCRIPTION
**Pull request by furkanipek**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description
Added notification provider using Bitrix24 Webhook.
Resolves #3610

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)
Uptime Kuma - Bitrix24 Settings

Bitrix24 - Notifications

### CODE REVIEW

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [inline server/notification-providers/bitrix24.js:12]:
```suggestion
        const okMsg = "Sent Successfully.";

        try {
            const params = {
```

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [inline server/notification-providers/bitrix24.js:5]:
```suggestion
```

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [inline server/notification-providers/bitrix24.js:29]:
```suggestion
            this.throwGeneralAxiosError(error);
```

File: src/components/notifications/Bitrix24.vue
Comment (CommanderStorm) [inline src/components/notifications/Bitrix24.vue:20]: Is adding the `div` with `input-group mb-3` intentional?

```suggestion
    <div class="mb-3">
        <label for="bitrix24-user-id" class="form-label">{{ $t("User ID") }}</label>
        <input id="bitrix24-user-id" v-model="$parent.notification.bitrix24UserID" type="text" class="form-control" required>
        <div class="form-text">{{ $t("bitrix24SupportUserID") }}</div>
    </div>
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
We also have testing and certificate-expiry notifications as noted in the issue.
Could you provide a screenshot of this case?
You can find expired certs at https://badssl.com/

**Comment by furkanipek** [comment]
DOWN/UP and testing/cert-expiry
@CommanderStorm Since I took the Telegram Notification files in Uptime Kuma as a reference, I added Bitrix24 without changing the structure.

I am sending screenshots.

**Review by CommanderStorm — APPROVED** [review]
Could you add the following to the description? (this way this issue is not left open once the PR is merged) 
Resolves #3610

**Comment by furkanipek** [comment]
> Could you add the following to the description? (this way this issue is not left open once the PR is merged) Resolves #3610

- [x] Ensure that the PR includes screenshots of the DOWN/UP and testing/cert-expiry events.
- [x] All keys that can/should be translated are in en.json (⇒ our translators can translate this and no merge-confilicts between Weblate and master can arise)

**Review by Computroniks — CHANGES_REQUESTED** [review]
Changes to the eslint rules in the 2.x branch require that all methods have JSDoc comments. Given that this is a notification provider and as such inherits from the base class, `@inheritdoc` is sufficient here.

**Comment by CommanderStorm** [comment]
Re: https://github.com/louislam/uptime-kuma/pull/3620#issuecomment-1687472262
No, I literally meant please add the string "Resolves #3610" to the [description of this Pull Request](https://github.com/louislam/uptime-kuma/pull/3620#issue-1858871758).
You can read more about [linking a pull request to an issue here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

File: server/notification-providers/bitrix24.js
Comment (chakflying) [inline server/notification-providers/bitrix24.js:14]: I don't think this is a good idea. Please use `heartbeatJSON["status"]` instead of `msg.indexOf`.

**Review by chakflying — CHANGES_REQUESTED** [review]

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [reply server/notification-providers/bitrix24.js:14]: Also note that `heartbeatJSON` might be `null`

**Comment by furkanipek** [comment]
> > Could you add the following to the description? (this way this issue is not left open once the PR is merged) Resolves #3610
> 
> * [x]  Ensure that the PR includes screenshots of the DOWN/UP and testing/cert-expiry events.
> * [x]  All keys that can/should be translated are in en.json (⇒ our translators can translate this and no merge-confilicts between Weblate and master can arise)

The image I uploaded was wrong, I wrote to inform you.

File: server/notification-providers/bitrix24.js
Comment (furkanipek) [reply server/notification-providers/bitrix24.js:14]: Here, the color we transmitted is displayed on the left side of the notification. I did it with green and red logic. I thought it was enough to be green if it was Up.

File: server/notification-providers/bitrix24.js
Comment (furkanipek) [reply server/notification-providers/bitrix24.js:7]: I didn't add it because I don't see it in other notification providers. Do I need to add it?

Comment (CommanderStorm) [reply server/notification-providers/bitrix24.js:10]: All other notification providers have this already applied on the v2 branch. See https://github.com/louislam/uptime-kuma/pull/3575

File: server/notification-providers/bitrix24.js
Comment (furkanipek) [reply server/notification-providers/bitrix24.js:17]: Thank you for the information. I will take this into consideration. Do you think I should make this addition for Bitrix24?

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [inline server/notification-providers/bitrix24.js:23]: Lets use this version and see if the linter does not like the line length

```suggestion
            await axios.get(`${notification.bitrix24WebhookURL}/im.notify.system.add.json`, { params });
```

**Review by CommanderStorm — APPROVED** [review]
Thanks for the new notification provider 🎉

All changes in this PR are small and uncontroversial
⇒ merging with junior maintainer approval

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [reply server/notification-providers/bitrix24.js:17]:
```suggestion
                "ATTACH[COLOR]": (heartbeatJSON ?? {})["status"] === UP ? "#b73419" : "#67b518",
```

File: server/notification-providers/bitrix24.js
Comment (CommanderStorm) [inline server/notification-providers/bitrix24.js:3]:
```suggestion
const { UP } = require("../../src/util");

```

**Pull request merged**
